package.json

```diff
@@ -34,13 +34,16 @@
     "@polkadot/rpc-provider": "^10.9.1",
     "@polkadot/util": "^12.4.2",
     "@polkadot/util-crypto": "12.4.2",
+    "@polkawatch/ddp-client": "^2.0.8",
     "@substrate/connect": "^0.7.31",
     "@zondax/ledger-substrate": "^0.41.2",
     "bignumber.js": "^9.1.2",
     "bn.js": "^5.2.1",
     "buffer": "^6.0.3",
     "chart.js": "^4.4.0",
+    "chroma-js": "^2.4.2",
     "date-fns": "^2.29.3",
+    "ellipsize": "^0.5.1",
```

Comment:
We can remove this dependency and use `ellipsisFn` from `@polkadot-cloud/utils` courtesy of @wirednkod. 

Full code and docs: https://github.com/paritytech/polkadot-cloud/blob/main/packages/utils/lib/index.ts#L466-L470

Reply:
@rossbulat I think ellipsisFN is broken.

Testing ellipsisFN('Canada',30,'right') and am getting "..." as result.

Same for 'United States' is returning 'Uni..."

I am going to apply the change anyway, and then look into this function. I don't know if there are unit tests, but probably would be best to just add some strange cases.

I will look into it later, unless you want to have a look first.

Reply:
Thank you @rvalle 
Based on our documentation, the 2nd argument defines `The amount of characters that the ellipsis will be`, meaning how many characters will appear, before the ellipsis is applied`.

Please note that there were some changes on the "left"/"right" (now "start"/"end") on latest release of `@polkadot-cloud/utils` v0.0.14 (2 days ago).

Though I did identified a bug thanks to you and I created: https://github.com/paritytech/polkadot-cloud/pull/598 which will result to v0.0.15.

Based on that latest release, when an amount of characters larger than the length of the string is given (while `start` or `end`), then this defaults to  `str.length - 3` + '...': 
Your code `ellipsisFN('Canada',30,'end')` which should result to: `Can...` and `ellipsisFN('United states',30,'end')` which should result to: `United sta...` 

If though you submit `ellipsisFN('United states',8 ,'end')` you should get: `United s...` while

Reply:
I merged main, with the .15 

@wirednkod @rossbulat I think the there is a case that ellipsisFN is not considering.

When the provided string size is smaller than N, then there is no ellipsis required and the string should be returned intact.

`
ellipsisFN("Malta", 13, "end") = "Ma..."
` 

However "Malta" is only 5, which is less than 13, and it should not need to be "ellipsized" anyway.  I suggest in that case the string is returned unchanged. I could do that check from outside but I believe it belongs in the function.

The thing is that the semantics of the library I used before and this one are different: previously you specify the number of characters you have space for, and the library will adjust the string as/if required.

My guess is that our ellipsisFN was traditionally used for Account Numbers, which are long and fixed-length, and therefore variable strings as input were not an use case.

Reply:
I have reviewed the ellipsisFN and I realize that is has been designed in a different way. Being used for long, constat size addresses the interface is designed to decide how you want the result to look, rather than using the UI space available.

Perhaps it is better that I deal with when to ellipsize/not ellipsize for now.

Reply:
I have now fixed this on my side, without needed to touch ellipsizeFN.

Reply:
Thank you @rvalle. The underlying concept of the "ellipse" function transcends mere automation of an ellipsis, which can be accomplished straightforwardly with CSS. Instead, it aims to provide to the user an ellipsis function that grants control over the number of characters to display, allowing for manual determination, and positioning the ellipsis (whether at the start, center, or end of the string). Initially conceived for handling the lengthy addresses of an account, this functionality though can be extended to other character strings as well.

The implementation includes specific provisions to address scenarios, where the character count designated as "to show", exceeds the length of the string (typically defaults to 4, if I remember correctly). More details on the usage, are available within the documentation page of the polkadot-cloud project. If there is a case that is not covered , I am more than willing to incorporate it. Please do not hesitate to create an issue in the [polkadot-cloud](https://www.github.com/paritytech/polkadot-cloud/issues) repository if such a situation arises.